src/core/hoc/ResizeDetectDiv.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import * as React from 'react';
+import { withResizeDetector } from 'react-resize-detector';
+
+interface ResizeDetectDivProps {
```

Comment:
I see that you are using types for checking the prop types. I think that this is nice, but it won't raise warnings when the consumer of the library is not using TS, or if there is a layer in-between not using TS. Have you considered that ? Is it much more maintenance to have the types plus the "real" proptypes ?

Reply:
I'll do some research on these! thanks for pointing in it out

Reply:
It's worth giving this a try 
https://fettblog.eu/typescript-react/prop-types/#inferring-prop-types

Reply:
I finally wrote it separately but would definitely go for the inferring for bigger ones